core/arch/aarchxx/mangle.c

```diff
@@ -1143,10 +1143,14 @@ reinstate_it_blocks(dcontext_t *dcontext, instrlist_t *ilist, instr_t *start,
             /* A label instruction may be used as a cti target, so we stop
              * the IT block on label instructions.
              */
-            !instr_is_label(instr) &&
-            /* Do not put OP_b exit cti into block: patch_branch can't handle */
-            instr_get_opcode(instr) != OP_b &&
-            instr_get_opcode(instr) != OP_b_short;
+            !instr_is_label(instr);
+        if (for_mangle) {
```

Comment:
Would it be preferable to not put *any* cti's into a block, something like

```asm
it ne
movne r1, r1
bne   ...
```

as opposed to

```asm
itt ne
movne r1, r1
bne   ...
```

I didn't think about this earlier, but it would at least be more symmetric wrt mangling and we wouldn't have to deal with all this `for_mangle` business (Does this make any sense? I know little about Thumb mode, but at least with gas this seems to work)